index.html

```diff
@@ -5020,180 +5020,154 @@ <h3 id="playback-AudioBufferSourceNode">
-          </ol>
+          <pre>
+// The AudioBuffer and context employed by this node
+var buffer;
+var context;
+
+// The following variables capture attribute and AudioParam values for the node.
+// They are updated k-rate, except for buffer.
+var loop;
+var detune;
+var loopEnd;
+var loopStart;
+var playbackRate;
+
+// State variables for the node's playback
+var start = 0;
+var offset;
+var stop = Infinity;
+
+var bufferTime = 0;
+var started = false;
+var enteredLoop = false;
+var dt;
+
+// Handle invocation of start method call
+function handleStart(when, pos, duration) {
+  if (arguments.length &gt;= 1) {
+    start = when;
+  }
+  offset = pos;
+  if (arguments.length &gt;= 3) {
+    stop = when + duration;
+  }
+}
+
+// Handle invocation of stop method call
+function handleStop(when) {
+  if (arguments.length &gt;= 1) {
+    stop = when;
+  }
+  else {
+    stop = context.currentTime;
+  }
+}
+
+// Interpolate a multi-channel signal value for some sample frame
+function playbackSignal(position) {
+  /*
+    This function provides the playback signal function for buffer, which is a
+    function that maps from a playhead position to a set of output signal
+    values, one for each output channel. If position corresponds to the
+    location of an exact sample frame in the buffer, this function returns
+    that frame. Otherwise, its return value is determined by a UA-supplied
+    algorithm that interpolates between sample frames in the neighborhood of
+    position.
+
+    If position is greater than or equal to loopEnd and there is no subsequent
+    sample frame in buffer, then interpolation should be based on the sequence
+    of subsequent frames beginning at loopStart.
+   */
+   ...
+}
+
+// Render a single frame of audio into the channel data arrays given by output
+// at the given element index. The frame value may be an array of channel signal
+// values, or a single value to be rendered to all channels.
+function renderFrame(output, index, frame) {
+  ...
+}
+
+// Generate a single render quantum of audio to be placed
+// in the channel arrays defined by output.
+function process(output) {
+  var dt = 1 / context.sampleRate;
+  var index = 0;
+
+  // Combine the two k-rate parameters affecting playback rate
+  var computedPlaybackRate = playbackRate * Math.pow(2, detune / 1200);
+
+  // Determine loop endpoints as applicable
+  var actualLoopStart, actualLoopEnd;
+  if (loop) {
+    if (loopStart &gt;= 0 &amp;& loopEnd &gt; 0 &amp;& loopStart &lt; loopEnd) {
+      actualLoopStart = loopStart;
+      actualLoopEnd = Math.min(loopEnd, buffer.duration);
+    }
+    else {
+      actualLoopStart = 0;
+      actualLoopEnd = buffer.duration;
+    }
+  }
+
+  // Render each sample frame in the quantum
+  while (index &lt; output[0].length) {
+    if (currentTime &lt; start || currentTime &gt;= stop) {
+      // emit silence for the output frame for the element specified by index
+      renderFrame(output, index, 0);
```

Comment:
To reduce nesting depth, it might be beneficial to have "continue" here so that the else statement below isn't needed?  I consider this to be the "exceptional" case, so quick exit allows me to concentrate the the heart of the algorithm.